portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -28,21 +28,46 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  // adding hard-coded messages to arraylist
   ArrayList<String> sampleMessages = new ArrayList<>();
 
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {  
     String messages = convertToJsonString(sampleMessages);
     response.setContentType("text/html;");
     response.getWriter().println(messages);
   }
 
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the inpiut from the form.
+    String userComment = getUserComment(request);
+    if(userComment == null) {
+        response.setContentType("text/html");
+        response.getWriter().println("Please enter a valid comment");
+        return;
+    }
+    sampleMessages.add(userComment);
+    response.sendRedirect("/dinosaur.html");
+  }
+
+  private String getUserComment(HttpServletRequest request) {
```

Comment:
you might want to add a comment to describe the function.

Reply:
okay